.github/workflows/ci.yml

```
@@ -1,134 +1,192 @@
         with:
           toolchain: nightly
-          override: true
 
-      - name: Run cargo-external-doc
-        uses: actions-rs/cargo@v1
+      - name: Install minimal dependencies versions
+        run: cargo +nightly update -Z minimal-versions
+
+      - run: cargo test --features testing-helpers
+
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu
+          - macOS
+          - windows
+        toolchain:
+          - stable
+          - nightly
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          command: test
-          args: --doc --features testing-helpers
+          toolchain: ${{ matrix.toolchain }}
+      - run: rustup default ${{ matrix.toolchain }}
 
+      - run: cargo test --features testing-helpers
 
-  clippy:
-    name: Clippy
+  test-features:
+    name: test features
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          profile: minimal
           toolchain: nightly
-          components: clippy
-          override: true
+      - run: rustup default nightly
 
-      - name: Clippy
-        uses: actions-rs/cargo@v1
-        with:
-          command: clippy
-          args: --all-features -- -D warnings
+      - name: Install tomljson
+        run: go get github.com/pelletier/go-toml/cmd/tomljson
 
+      - run: ci/test_all_features.sh
+        env:
+          GOBIN: "$HOME/go/bin"
+          PATH: "$PATH:$GOBIN"
 
-  rustfmt:
+
+
+
+  #################
+  # Documentation #
+  #################
+
+  rustdoc:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
         with:
-          profile: minimal
           toolchain: nightly
-          components: rustfmt
-          override: true
 
-      - name: rustfmt
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+      - run: cargo +nightly doc
+        env:
+          RUSTDOCFLAGS: --cfg docsrs
 
 
-  msrv:
-    # TODO: Use MSRV once it understands new `Cargo.lock` format.
-    # name: MSRV
-    name: Minimal crates versions
-    strategy:
-      fail-fast: false
-      matrix:
-        # msrv: ["1.36.0"]
-        msrv: [stable]
-        os: [ubuntu, macOS, windows]
-    runs-on: ${{ matrix.os }}-latest
+
+
+  #############
+  # Releasing #
+  #############
+
+  release-crate:
+    name: release on crates.io
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+    needs: ["release-github"]
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v3
+      - uses: dtolnay/rust-toolchain@v1
```

based on the docs for this action the `@rev` should include the rust version: 

```suggestion
      - uses: dtolnay/rust-toolchain@stable
```
Source: https://github.com/dtolnay/rust-toolchain

---

@JelteF nope, it works perfectly well either way. The `dtolnay/rust-toolchain@stable` syntax is just a neat shorthand. However, I do prefer not to use it, as I'm unsure that @dependabot will understand it correctly and propose updates once new major version is out.